packages/react-server/src/render/render.tsx

```diff
@@ -133,8 +148,16 @@ export function createRender(render: RenderFunction, options: Options = {}) {
         assets.scripts({name: assetName, asyncAssets: immediateAsyncAssets}),
       ]);
 
+      styles.push(...additionalStyles);
```

Comment:
Making sure that if somebody decides to use styles/scripts then they don't clobber the assets provided by `getAssets`. I've added a test case for this.